src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -4372,15 +4474,20 @@ private ImmutableArray<TypeWithAnnotations> InferMethodTypeArguments(
             // a possibly constructed generic type, is exceedingly subtle. See the comments
             // in "Infer" for details.
 
+            PooledDictionary<TypeParameterSymbol, int> ordinals = makeOrdinalsIfNeeded(member, originalTypeParameters);
+
             var inferenceResult = MethodTypeInferrer.Infer(
                 _binder,
                 _binder.Conversions,
                 originalTypeParameters,
-                method.ContainingType,
+                getContainingType(member),
```

Comment:
>getContainingType(member) [](http://example.com/codeflow?start=16&length=25)

Why old value didn't work?

Reply:
The old value works, but it feels wrong.
It just feels weird to feed a symbol that is being inferred, as opposed to the actual containing type.
The two usage spots are:
`_fixedResults[i] = (TypeWithAnnotations.Create(new ExtendedErrorTypeSymbol(_constructedContainingTypeOfMethod, _methodTypeParameters[i].Name, 0, null, false)), false);`
`TypeMap typeMap = new TypeMap(_constructedContainingTypeOfMethod, _methodTypeParameters, fixedArguments);` (note that `ForType` does inspect the type in case it's a substituted type)